### PR TITLE
Sentry integration

### LINKED ISSUE #291: Experiment with Sentry integration
good place to start: https://docs.sentry.io/platforms/dotnet/guides/log4net/

### PR DESCRIPTION
**Pull request by IhateTrains**
closes #291

### CODE REVIEW

File: Fronter.NET/Services/ConverterLauncher.cs [hunk 1]
```diff
@@ -115,16 +119,109 @@ internal class ConverterLauncher {
...
+			var message = $"Converter exited with code {processExitCode}";
+			SentrySdk.CaptureMessage(message, SentryLevel.Error);
+		}
+	}
+
+	private static async Task UploadSaveArchiveToBackblaze(string archivePath) {
+		// Init Backblaze B2 client.
+		var client = new BackblazeClient();
+		const string keyId = "0030b6343d5e7b30000000001";
+		const string applicationKey = "K003NNlYJwOJQW0YmxY7ZmMBJekoyJM";
```
[the remark this answers is not in the document]
Comment (IhateTrains) [reply Fronter.NET/Services/ConverterLauncher.cs:208]: Will generate a new key and put it into GitHub secrets.

File: Fronter.NET/Services/ConverterLauncher.cs [hunk 2]
```diff
@@ -115,16 +119,109 @@ internal class ConverterLauncher {
...
+		
+		// Create zip with save file.
+		var dateTimeString = DateTime.Now.ToString("yyyy-MM-dd_HH-mm-ss");
+		var archivePath = $"temp/SaveGame_{dateTimeString}.zip";
+		using (var zip = ZipFile.Open(archivePath, ZipArchiveMode.Create)) {
+			zip.CreateEntryFromFile(saveLocation, new FileInfo(saveLocation).Name);
+		}
+
+		var archiveSize = new FileInfo(archivePath).Length; // Size in bytes.
+		if (archiveSize <= 19 * 1024 * 1024) {
```
Comment (Idhrendur) [inline Fronter.NET/Services/ConverterLauncher.cs:179]: Every so often a log file gets absurdly large. Why not measure the log file size and use that to determine the limit?

**Review by Idhrendur — APPROVED** [review]

File: Fronter.NET/Services/ConverterLauncher.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (IhateTrains) [reply Fronter.NET/Services/ConverterLauncher.cs]: The credentials are now added from env variables to a Secrets class in a custom build target:
```
    <!-- https://stackoverflow.com/a/50905092/10249243 -->
    <Target Name="Secrets" BeforeTargets="BeforeBuild">
        <WriteLinesToFile 
            File="$(IntermediateOutputPath)gen.cs"
            Lines="static partial class Secrets {
			    public static string BackblazeKeyId = %22$(BACKBLAZE_KEY_ID)%22 %3B
			    public static string BackblazeApplicationKey = %22$(BACKBLAZE_APPLICATION_KEY)%22 %3B
			    public static string BackblazeBucketId = %22$(BACKBLAZE_BUCKET_ID)%22 %3B
			}"
			Overwrite="true"
        />
	</Target>
	<ItemGroup>
		<Compile Include="$(IntermediateOutputPath)gen.cs" />
	</ItemGroup>
```

File: Fronter.NET/Services/ConverterLauncher.cs (hunk 2 above)
Comment (IhateTrains) [reply Fronter.NET/Services/ConverterLauncher.cs]: Makes sense. Now I check whether the save archive is <= `20 MB - <save size> - <arbitrary 0.5 MB for base Sentry request>.

**Pull request merged**